src/RpcClient.ts

```diff
@@ -354,4 +372,119 @@ export class RpcClient {
       throw new Error(`Error in getSignaturesForAsset: ${error}`);
     }
   }
+  /**
+   * Mint a compressed NFT.
+   * @returns {Promise<MintCNFTResponse>}
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResponse | undefined> {
+    try {
+      const ownerKeypair = request.ownerKeypair;
+      // User can override defaulted values.
+      const defaultMetadataArgs: RequiredMetadataArgs = {
+        primarySaleHappened: false,
+        isMutable: false,
+        editionNonce: 0,
+        tokenStandard: null,
+        uses: null,
+        tokenProgramVersion: null,
+        ...request.metadataArgs,
+      };
+      let assetId = "";
+
+      let treeWallet = request.treeKeypair ?? Keypair.generate();
+      if (!request.treeKeypair) {
+        await initTree(this.connection, ownerKeypair, treeWallet);
+      }
+      let response: any;
+      let collection = request.collection ?? false;
+
+      // User set collection to false, or it was not dictated.
+      if (collection == false && treeWallet) {
+        const signature = await mintCompressedNFT(
+          this.connection,
+          defaultMetadataArgs,
+          request.ownerKeypair,
+          treeWallet
+        );
+        const { id } = await getAssetID(
+          this.connection.rpcEndpoint,
+          treeWallet.publicKey,
+          signature
+        );
+        response = {
+          assetId: id,
+          creator: request.ownerKeypair.publicKey.toBase58(),
+          treeId: treeWallet.publicKey.toBase58(),
+          treeKeypair: treeWallet.secretKey,
+          signature: signature,
+        };
+        assetId = id;
+        // User wants a collection
+      } else if (collection == true) {
+        let collectionMint = request.collectionMint;
+        // User did not supply a public key for exisiting collection
+        if (!collectionMint) {
+          const collectionInfo = await initCollection(
+            this.connection,
+            ownerKeypair,
+            request.metadataArgs.name,
+            request.metadataArgs.symbol,
+            request.metadataArgs.uri
+          );
+          collectionMint = collectionInfo.collectionMint;
+          // User did supply a collection mint Public key.
```

Comment:
If `collection == true` and `collectionMint` is not provided by the user, this will assign a new address to the `collectionMint` and simply exit without minting the cNFT. To address this, we should merge the next `if else` block with this block. 

```ts
else if (collection == true) {
        let collectionMint = request.collectionMint;
        // User did not supply a public key for exisiting collection
        if (!collectionMint) {
          // do stuff
          collectionMint = collectionInfo.collectionMint;
        }
        
        // go on as if user provided collectionMint, now that we have one
        const { collectionMetadataAccount, collectionMasterEditionAccount } = await existingCollection(collectionMint);
          const signature = await mintCollectionCompressedNft(
          // ...
```